### PR TITLE
Consolidate test installer to use build-artifacts-in-docker

### PR DESCRIPTION
**Pull request by petroav**

### CODE REVIEW

File: bin/build-artifacts-in-docker.sh [hunk 1]
```diff
@@ -1,37 +1,98 @@
 #!/usr/bin/env bash
 
-set -e
-set -o pipefail
-set -x
+set -eox pipefail
 
-ROOT_DIR=$(readlink -f $(dirname $0)/..)
+function usage() {
```
Comment (anusudarsan) [inline bin/build-artifacts-in-docker.sh:5]: 🥇

File: bin/build-artifacts-in-docker.sh [hunk 2]
```diff
@@ -1,37 +1,98 @@
...
 fi
 
-BASE_IMAGE_NAME=${BASE_IMAGE_NAME}-build
-
 if [[ -z "${BASE_IMAGE_TAG}" ]]; then
   BASE_IMAGE_TAG=$(cat ${ROOT_DIR}/base-images-tag.json | python -c 'import sys, json; print json.load(sys.stdin)["base_images_tag"]')
 fi
 
+if [[ -z "${DIST_TARGETS}" ]]; then
+  DIST_TARGETS="dist-online dist-offline"
```
Comment (anusudarsan) [inline bin/build-artifacts-in-docker.sh:70]: we have a `dist-online` target same as `dist`?

File: tests/product/prestoadmin_installer.py [hunk 3]
```diff
@@ -82,69 +82,35 @@ def _build_installer_in_docker(self, cluster, online_installer=None,
...
-                'pip install --upgrade setuptools==20.1.1\n'
-                'mv %s/presto-admin ~/\n'
-                'cd ~/presto-admin\n'
-                'make %s\n'
-                'cp dist/prestoadmin-*.tar.gz %s'
-                % (installer_container.mount_dir,
-                   'dist' if online_installer else 'dist-offline',
-                   installer_container.mount_dir),
-                container_name)
+            cmd = [prestoadmin.main_dir + "/bin/build-artifacts-in-docker.sh",
```
Comment (anusudarsan) [inline tests/product/prestoadmin_installer.py:94]: 👍

File: tests/product/prestoadmin_installer.py [hunk 4]
```diff
@@ -82,69 +82,35 @@ def _build_installer_in_docker(self, cluster, online_installer=None,
...
             installer_file = fnmatch.filter(
-                os.listdir(local_container_dist_dir),
+                os.listdir(os.path.join(temp_dir, "dist")),
                 'prestoadmin-*.tar.gz')[0]
             shutil.copy(
-                os.path.join(local_container_dist_dir, installer_file),
+                os.path.join(os.path.join(temp_dir, "dist"), installer_file),
                 cluster.get_dist_dir(unique))
         finally:
-            installer_container.tear_down()
```
Comment (anusudarsan) [inline tests/product/prestoadmin_installer.py:147]: is the `tear_down` used elsewhere? can we remove that as well?

File: bin/build-artifacts-in-docker.sh [hunk 5]
```diff
@@ -1,37 +1,98 @@
...
+  env CONTAINER_DIR="${CONTAINER_DIR}" env DIST_TARGETS="${DIST_TARGETS}" bash <<"EOF"
     set -x
     cd ${CONTAINER_DIR}
-    pip install --ignore-installed -r requirements.txt
-    pip install tox-travis==0.10
-    # list presto-admin package dependencies versions
-    pip install Babel==2.5.3
-    pip install cffi==1.11.5
-    pip install PyNaCl==1.2.1
-    pip install cryptography==2.1.1
```
Comment (anusudarsan) [inline bin/build-artifacts-in-docker.sh:34]: why did you remove requirements.txt ?

File: tests/product/prestoadmin_installer.py [hunk 6]
```diff
@@ -82,69 +82,35 @@ def _build_installer_in_docker(self, cluster, online_installer=None,
...
-                'pip install --upgrade pycparser==2.18 idna==2.1 cryptography==2.1.1\n'
-                'pip install --upgrade pip==9.0.2\n'
-                'pip install --upgrade wheel==0.23.0\n'
-                'pip install --upgrade setuptools==20.1.1\n'
-                'mv %s/presto-admin ~/\n'
-                'cd ~/presto-admin\n'
-                'make %s\n'
-                'cp dist/prestoadmin-*.tar.gz %s'
-                % (installer_container.mount_dir,
-                   'dist' if online_installer else 'dist-offline',
```
Comment (anusudarsan) [inline tests/product/prestoadmin_installer.py:127]: so we never ran tests in `online_installer` here which looks like some dead code https://github.com/prestosql/presto-admin/pull/22/files#diff-0e9360ea0e4c1e02b85edc7c23fa0984R82 ? I dont want to clean all that up, just asking

File: bin/build-artifacts-in-docker.sh [hunk 7]
```diff
@@ -1,37 +1,98 @@
...
+  [--base_image_name <name>]: Name of Docker image to use during the build. Defaults to prestodev/centos6-presto-admin-tests.
+  [--base_image_tag <tag>]:   Tag of the Docker image to use during the build. Defaults to the tag stored in the base_images_tag.json file.
+  [--online-dist]:            Boolean argument indicating whether the online distribution should be build. If neither the
+                              --online-dist nor the --offline-dist flags are specified then both distributions are built.
+  [--offline-dist]:           Boolean argument indicating whether the offline distribution should be build. If neither the
+                              --online-dist nor the --offline-dist flags are specified then both distributions are built.
+EOF
+}
+
+while [[ $# -gt 0 ]]
```
Comment (findepi) [inline bin/build-artifacts-in-docker.sh:21]: by convention, in bash scripts we use getopt, but AFAIK it supportd short options only
(for anything that is going really complex, bash isn't the tool anyway)

it's fine here to do "manual" opt parsing -- this is just test infrastructure script, so supporting `--opt arg` but not supporting `--opt=arg` is OK _here_. for production use we would need to use proper library (or proper language)

File: tests/product/test_server_install.py [hunk 8]
```diff
@@ -219,153 +224,42 @@ def test_install_ext_host_is_pa_master(self):
...
-            self.cluster.master
+        self.upload_topology(
+            {"coordinator": ips[self.cluster.internal_master],
+             "workers": [ips[self.cluster.internal_slaves[0]],
+                         ips[self.cluster.internal_slaves[1]],
+                         ips[self.cluster.internal_slaves[2]]],
+             "username": "app-admin"}
         )
 
-        cmd_output = installer.install().splitlines()
```
Comment (anusudarsan) [inline tests/product/test_server_install.py:272]: 👍

File: bin/build-artifacts-in-docker.sh (hunk 1 above)
Comment (petroav) [reply bin/build-artifacts-in-docker.sh:5]: :pray:

File: bin/build-artifacts-in-docker.sh (hunk 2 above)
Comment (petroav) [reply bin/build-artifacts-in-docker.sh:70]: Ya, `dist` is just a link to `dist-online`.

File: tests/product/prestoadmin_installer.py (hunk 4 above)
Comment (petroav) [reply tests/product/prestoadmin_installer.py:147]: `tear_down` is a method on the `DockerCluster` class so it is used in a bunch of places.

File: bin/build-artifacts-in-docker.sh (hunk 5 above)
Comment (petroav) [reply bin/build-artifacts-in-docker.sh:34]: `requirements.txt` is used for setting up your development env so it contains dependencies that are not relevant for building the installer.

File: tests/product/prestoadmin_installer.py (hunk 6 above)
Comment (petroav) [reply tests/product/prestoadmin_installer.py:127]: The online installer was definitely exercised. Take a look at `test_online_installer.py`, which calls this method directly with `online_installer=True`. However, I don't know if the particular IF statement you link to ever gets exercised.

File: bin/build-artifacts-in-docker.sh [hunk 9]
```diff
@@ -1,37 +1,98 @@
 #!/usr/bin/env bash
 
-set -e
-set -o pipefail
-set -x
+set -eox pipefail
```
[the remark this answers is not in the document]
Comment (petroav) [reply bin/build-artifacts-in-docker.sh:3]: Done

File: bin/build-artifacts-in-docker.sh [hunk 10]
```diff
@@ -1,37 +1,98 @@
...
 
-ROOT_DIR=$(readlink -f $(dirname $0)/..)
+function usage() {
+  cat << EOF
+build-artifacts-in-docker.sh [--root_dir <dir>] [--base_image_name <name>] [--base_image_tag <tag>] [--online-dist] [--offline-dist]
+
+This script is used to build the offline and/or online distributions of presto-admin in the specified Docker image name.
+  [--root_dir <dir>]:         Directory on local filesystem where the presto-admin tree is located. Defaults to this
```
[the remark this answers is not in the document]
Comment (petroav) [reply bin/build-artifacts-in-docker.sh:10]: Removed opening and closing square brackets from the argument listing

File: bin/build-artifacts-in-docker.sh [hunk 11]
```diff
@@ -1,37 +1,98 @@
...
+  [--online-dist]:            Boolean argument indicating whether the online distribution should be build. If neither the
+                              --online-dist nor the --offline-dist flags are specified then both distributions are built.
+  [--offline-dist]:           Boolean argument indicating whether the offline distribution should be build. If neither the
+                              --online-dist nor the --offline-dist flags are specified then both distributions are built.
+EOF
+}
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
```
[the remark this answers is not in the document]
Comment (petroav) [reply bin/build-artifacts-in-docker.sh:23]: Done
